mutation_query.cc

```diff
@@ -76,3 +77,20 @@ auto fmt::formatter<reconcilable_result::printer>::format(
 reconcilable_result::printer reconcilable_result::pretty_printer(schema_ptr s) const {
     return { *this, std::move(s) };
 }
+
+lw_shared_ptr<reconcilable_result> reversed(lw_shared_ptr<reconcilable_result>&& result)
+{
+    return ::make_lw_shared(reversed(std::move(*result)));
+}
+
+reconcilable_result reversed(reconcilable_result&& result)
+{
+    for (auto& partition : result.partitions())
+    {
+        auto& m = partition.mut();
+        auto schema = local_schema_registry().get(m.schema_version());
+        m = frozen_mutation(reverse(m.unfreeze(schema)));
```

Comment:
This has the potential to stall.

If we have similar O(n) operations on reconcilable_result, then we can say it isn't making it worse, but if other reconcilable_result operations are careful to yield, then so must this.